Floofbot/Services/NicknameAlertService.cs

```diff
@@ -0,0 +1,174 @@
+﻿using Discord;
+using Discord.Addons.Interactive;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Floofbot.Services
+{
+    public class NicknameAlertService : InteractiveBase
+    {
+        private FloofDataContext _floofDb;
+        
+        private Dictionary<ulong, SocketGuildUser> messageDic = new Dictionary<ulong, SocketGuildUser>(); 
+        private ITextChannel _channel;
+        private Emoji banEmoji = new Emoji("🔨");
+        private Emoji warnEmoji = new Emoji("⚠️");
+        private Emoji kickEmoji = new Emoji("👢");
+        private Emoji removeEmoji = new Emoji("📝");
+
+        public NicknameAlertService(FloofDataContext floofDb)
+        {
+            _floofDb = floofDb;
+        }
+        public async Task<ITextChannel> GetChannel(Discord.IGuild guild, ulong channelId)
+        {
+            return await guild.GetTextChannelAsync(channelId);
+        }
+
+        public async Task HandleBadNickname(SocketGuildUser badUser, IGuild guild)
+        {
+            var serverConfig = _floofDb.NicknameAlertConfigs.Find(guild.Id);
+
+            if (serverConfig == null || !serverConfig.IsOn || serverConfig.Channel == 0) // not configured/disabled
+            {
+                return;
+            }
+            _channel = await GetChannel(guild, serverConfig.Channel);
+
+            var embed = new EmbedBuilder()
+                .WithDescription($"{removeEmoji.Name}: Remove Nickname\n" +
+                $"{warnEmoji.Name}: Warn\n" +
+                $"{kickEmoji.Name}: Kick\n" +
+                $"{banEmoji.Name}: Ban")
+                .Build();
+
+            var message = await _channel.SendMessageAsync($"{badUser.Mention} ({badUser.Username}#{badUser.Discriminator}) has been " +
+                $"detected with a bad name! What should I do?\n\nNickname: {badUser.Nickname}", false, embed);
+            await message.AddReactionAsync(removeEmoji);
+            await message.AddReactionAsync(kickEmoji);
+            await message.AddReactionAsync(warnEmoji);
+            await message.AddReactionAsync(banEmoji);
+
+            messageDic.Add(message.Id, badUser);
+
+        }
+
+        public async Task ReactionCallback(Cacheable<IUserMessage, ulong> message, ISocketMessageChannel channel, SocketReaction reaction)
```

Comment:
Would be clearer if it was like an event name such as OnReactionPressed

Reply:
the event in the api is called ReactionAdded so renamed to OnReactionAdded